Test file for the function say_my_name() in 3-say_my_name.py
=====================================================================

Module ``say_my_name``
---------------------

Import the function:
    >>> say_my_name = __import__('3-say_my_name').say_my_name

-----------------------------
-----------------------------
    The function should work like this

    >>> say_my_name("John", "Smith")
    My name is John Smith

-----------------------------
-----------------------------
    last_name is missing

    >>> say_my_name("Bob")
    My name is Bob 

-----------------------------
-----------------------------
    first_name is not a string

    >>> say_my_name(28, "Capote")
    Traceback (most recent call last):
    TypeError: first_name must be a string

    >>> say_my_name(None)
    Traceback (most recent call last):
    TypeError: first_name must be a string

-----------------------------
-----------------------------
    last_name is not a string

    >>> say_my_name("Dina", 3.5)
    Traceback (most recent call last):
    TypeError: last_name must be a string

    >>> say_my_name("Dina", None)
    Traceback (most recent call last):
    TypeError: last_name must be a string

-----------------------------
-----------------------------
    Incorrect amount of parameters

    >>> say_my_name()
    Traceback (most recent call last):
    TypeError: say_my_name() missing 1 required positional argument: 'first_name'

    >>> say_my_name("Dina", "Capote", "Alfonso")
    Traceback (most recent call last):
    TypeError: say_my_name() takes from 1 to 2 positional arguments but 3 were given
